packages/server/modules/core/loaders.js

```diff
@@ -0,0 +1,58 @@
+const _ = require('lodash')
+const { StreamFavorites, knex } = require('@/modules/core/dbSchema')
+const DataLoader = require('dataloader')
+
+/**
+ * All DataLoaders available on the GQL ctx object
+ * @typedef {Object} RequestDataLoaders
+ * @property {{
+ *  getUserFavoriteData: DataLoader<string, Object>,
+ *  getFavoritesCount: DataLoader<string, number>
+ * }} streams
+ */
+
+module.exports = {
+  /**
+   * Build request-scoped dataloaders
+   * @param {Object} ctx GraphQL context
+   * @returns {RequestDataLoaders}
+   */
+  buildRequestLoaders(ctx) {
+    const userId = ctx.userId
+
+    return {
+      streams: {
+        /**
+         * Get favorite metadata for a specific stream and user
+         */
+        getUserFavoriteData: new DataLoader(async (keys) => {
```

Comment:
this is neat :)
how can u run tests on this withour relying on the graphql server?
U'd have to extract the StreamsFavorites by streamId query part into a separate function, and test that right?